util/yamlinternal/decode.go

```diff
@@ -0,0 +1,60 @@
+package yamlinternal
+
+import (
+	"fmt"
+	"reflect"
+
+	_ "unsafe"
+
+	"gopkg.in/yaml.v3"
+)
+
+// decoder is a copy from gopkg.in/yaml.v3/decode.go
```

Comment:
Thanks for adding this comment!

That's Apache and this is MIT, so it wouldn't hurt to add headers to this file like

```
//
// Copyright (c) 2024 RELEX Oy
// Copyright (c) 2011-2019 Canonical Ltd
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```